### DISCUSSION TITLE
how to get unregistered fields from initial state in onSubmit function

### DISCUSSION THREAD

**Discussion by soojison** [react-hook-form/react-hook-form#2564] [Q&A]
Hey all, I have a problem I've been banging my head on for a while. I am using fieldArray for this problem, but I don't think this is a field array issue.

Say I have a payload that was fetched from the backend that looks like:
```
jobs: [{label: 'abc', id: '234'}, {label: 'def', id: '567'}]
```

In my code, I have only the label field registered like so:
```
<input
    name={`jobs[${index}].label`}
    defaultValue={item.label}
    ref={register({ required: true })}
/>
```

and when I click the submit button, the values I get in the `onSubmit` function contains only the `label` field, as expected. 
```
jobs: [{label: 'abc'}, {label: 'def'}]
```
However, I also want to get the `id` fields from the payload passed in this function. Is there a way to access these values somehow? 

I can't use the `find` method in the initial state array, as the user could change the label field.

**Comment by bluebill1049** [comment]
you can use <input type="hidden" /> or merge data at submit.

**Reply by soojison** [reply]
Is there a specific method to hook form I can use, or do I have to merge them manually?

**Reply by bluebill1049** [reply]
Another solution is to put your data in hidden inputs.
